nixos/modules/services/networking/knot.nix

```diff
@@ -5,14 +5,16 @@ with lib;
 let
   cfg = config.services.knot;
 
-  configFile = pkgs.writeText "knot.conf" cfg.extraConfig;
-  socketFile = "/run/knot/knot.sock";
+  configFile = pkgs.writeTextFile {
```

Comment:
I tried to convert nix attrsets to yaml using `remarshal`. However the yaml subset chosen by knot is too restrictive for that (i.e. expect keys to appear in a certain order).

Reply:
Have you seen [this thread](https://github.com/NixOS/nixpkgs/pull/56922#issuecomment-470740544)?  If a few people tries this out in practice, I suppose we could make it available as another optional attribute.

Reply:
I'm not aware of keys being expected in particular order, though (perhaps I misunderstand).

Reply:
I had an issue with an acl rule where the `id` key was not the first key, which lead to an error. This broke already in the nixos test. Anyway this PR should not interfere with migrating to a different contribute type...

Reply:
Oh, that's a known restriction, though not by me until now.

Reply:
I think I have an idea how to approach this; hopefully I can get it to work soon.  I'll link from here when that happens.

Reply:
It didn't really go the way I imagined, but let's continue on https://github.com/NixOS/nixpkgs/pull/81460